app/directives/megasearch.js

```diff
@@ -11,6 +11,14 @@ angular
 			},
 			link: function(scope, elem) {
 				scope.provideSuggestions = function(keyCode) {
+
+					// size the mega search box to match the input field
+					if($location.path()==='/'){
+						$('#search-terms #megasearchbox').width($('#search-terms #searchbox').width()-18);
```

Comment:
I am using the same technique for getting the location path as other existing code in the same site. (see basket.js, slimming, advanced filters, facetSearch etc) 

If you want this recoded, then we have a new and wider issue to tackle